DM-33999: Use unicode in run names for some tests

timj opened the pull request:
## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`

timj committed:
Factor out default run name in test

Define "ingest" in one place.

timj committed:
Relax regex to alphanumerics instead of ASCII alphanumerics

This allows for unicode collection names.

timj committed:
Use unicode in some test run names

timj committed:
Add support for emoji in collection names

timj commented on python/lsst/daf/butler/core/utils.py:
```diff
@@ -103,7 +103,8 @@ def globToRegex(
 
-    nomagic = re.compile(r"^[\w/\.\-@]+$", re.ASCII)
+    # The character class range at the end adds support for emoji.
+    nomagic = re.compile(r"^[\w/\.\-@\u263a-\U0001f645]+$")
```
This is a little brute force in terms of character ranges. There are more limited ranges that can be supported for emoticons and flags etc rather than allowing all emoji.

ktlim replied:
Seems to me that you should give up on trying to identify legal characters and instead merely detect whether there are glob characters that need `fnmatch`:
```
nomagic = re.compile(r"^[^*?[\]]*$")
```

timj replied:
Okay. That is probably better. This is specifically designed for `fnmatch` -- I wasn't entirely sure of the full list but I'll assume the above is correct.

ktlim replied:
I took the list from https://docs.python.org/3/library/fnmatch.html

ktlim replied:
In fact `fnmatch` doesn't seem to treat a `]` as special unless it's paired with `[`, so you could leave the `\]` out.

ktlim replied:
Oh, and you could invert the logic then, making it just an RE to look for one of the three characters (`*?[`) anywhere in the string.

timj replied:
Yes, I'm inverting.

timj committed:
Add news fragment

timj committed:
Be explicit about the fnmatch glob characters

This is more general than trying to list all allowed
characters that are not special.

See https://docs.python.org/3/library/fnmatch.html

Also improve the tests to explicitly test these
glob options.
